### PR TITLE
fix for help RPC call requiring passphrase

### PR DESCRIPTION
**Pull request by riordant**
Fixes https://github.com/zcoinofficial/zcoin/issues/672

## PR intention
The result of the `help` call with a locked & encrypted wallet is currently:
```
Error: Please enter the wallet passphrase with walletpassphrase first.
```
The issue is that the function `regeneratemintpool` does not have a `help` section, but requires wallet unlock. This PR adds that and makes the `help` function accessible from a locked wallet again.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2654,6 +2654,11 @@ UniValue fundrawtransaction(const UniValue& params, bool fHelp)
 
 UniValue regeneratemintpool(const UniValue &params, bool fHelp) {
 
+        if (fHelp || params.size() > 0)
+	        throw runtime_error(
+       	        	"regeneratemintpool\n"
+       		        "\nIf issues exist with the keys that map to mintpool entries in the DB, this function corrects them.\n"
+                );
```
Comment (ultimaweapon) [inline src/wallet/rpcwallet.cpp]: We should provides examples too so it will be the same as other Bitcoin RPCs.

Comment (riordant) [reply src/wallet/rpcwallet.cpp]: I had considered that but no arguments are being passed. What do you suggest in this case?

Comment (ultimaweapon) [reply src/wallet/rpcwallet.cpp]: We should show it without arguments like `getrawchangeaddress`.

Comment (riordant) [reply src/wallet/rpcwallet.cpp]: Updated to include this.

**Pull request merged**